transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -707,16 +728,7 @@ def _parse_v1_scan_response(self, sender, scan_data):
         info['current_time'] = curr_time
         info['last_seen'] = datetime.datetime.now()
 
-        self._trigger_callback('on_scan', self.id, info, self.ExpirationTime)
-
-        # If there is a valid reading on the advertising data, broadcast it
-        if stream != 0xFFFF:
-            if self._throttle_broadcast and self._check_update_seen_broadcast(sender, reading_time, stream, reading):
-                return
-
-            reading = IOTileReading(reading_time, stream, reading, reading_time=datetime.datetime.utcnow())
-            report = BroadcastReport.FromReadings(info['uuid'], [reading], curr_time)
-            self._trigger_callback('on_report', None, report)
+        return info, reading_time, stream, reading
```

Comment:
Same comment here. We should document why it was necessary to factor out this functionality and be very careful about changing logic around this part of the code.

Reply:
It's not necessary, my intention was  to handle callback in one place.